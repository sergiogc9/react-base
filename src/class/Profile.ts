export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
