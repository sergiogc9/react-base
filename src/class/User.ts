export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
