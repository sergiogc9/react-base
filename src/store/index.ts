import { createBrowserHistory } from 'history';
import isEqual from 'lodash/isEqual';
import { combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { all } from 'redux-saga/effects';

import { connectRouter } from 'connected-react-router';

import { connectApi } from '@src/lib/ajax/Api';
import { getMiddleware, sagaMiddleware } from '@src/middleware';
import { State } from './types';

import * as app from './app';

const INITIAL_STATE: State = {
	app: app.INITIAL_STATE,
	router: { action: "POP", location: { hash: "", pathname: "/", search: "", state: undefined } }
};

// define sagas generator function to register all sagas
function* sagas() {
	yield all([
		app.sagas()
	]);
}

const history = createBrowserHistory();
// patch to avoid stacking same location multiple times in history
const historyPush = history.push;
let lastLocation = history.location;
history.listen(location => lastLocation = location);
history.push = (path: any, state: object = {}): void => {
	const lastPath = lastLocation.pathname + lastLocation.hash + lastLocation.search;
	if (lastLocation === null || path !== lastPath || !isEqual(state, lastLocation.state)) historyPush(path, state);
};

// create reducers

const reducers = combineReducers({
	app: app.reducers,
	router: connectRouter(history)
});

const store = createStore(reducers, composeWithDevTools(getMiddleware(history)));
connectApi(store);

sagaMiddleware.run(sagas);

export { INITIAL_STATE, sagas, history, reducers, store };
