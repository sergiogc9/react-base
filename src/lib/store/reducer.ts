import { CaseReducer, createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit';

import { EntitiesFetchData } from 'store/entities';

// General reducer stuff
type CaseReducerWithPrepare<State, Payload> = {
	reducer: CaseReducer<State, {
		payload: Payload;
		type: string;
	}>;
	prepare: (payload: Payload) => {
		payload: Payload;
		meta: object | undefined;
	};
}

const getReducer = <State, Payload = void>(reducer: CaseReducer<State, PayloadAction<Payload>>, meta?: object): CaseReducerWithPrepare<State, Payload> => ({
	reducer,
	prepare: (payload: Payload) => ({ payload, meta })
});

// Api specific reducer stuff
export type DefaultPayloadError = { code: string, message: string };

type ApiOperatorOptions = {
	reload?: boolean // Used to reload page when optimistic api call fails
};

type ApiReducers<State, StartPayload, SuccessPayload, ErrorPayload> = {
	start?: CaseReducer<State, PayloadAction<StartPayload>>
	success?: CaseReducer<State, PayloadAction<SuccessPayload>>
	error?: CaseReducer<State, PayloadAction<ErrorPayload>>
}

const getApiReducers = <State, StartPayload = void, SuccessPayload = void, ErrorPayload = DefaultPayloadError>(
	reducers: ApiReducers<State, StartPayload, SuccessPayload, ErrorPayload>,
	options?: ApiOperatorOptions
): [
		CaseReducerWithPrepare<State, StartPayload>,
		CaseReducerWithPrepare<State, SuccessPayload>,
		CaseReducerWithPrepare<State, ErrorPayload>
	] => {
	const defaultReducer: CaseReducer<State> = state => state as State;
	return [
		getReducer<State, StartPayload>(reducers.start || defaultReducer, { api: 'start' }),
		getReducer<State, SuccessPayload>(reducers.success || defaultReducer, { api: 'success' }),
		getReducer<State, ErrorPayload>(reducers.error || defaultReducer, { api: 'error', reload: options?.reload })
	];
};

const getEntityFetchReducers = <T, StartPayload = void, SuccessPayload = void, ErrorPayload = DefaultPayloadError>() => {
	return getApiReducers<EntityState<T> & EntitiesFetchData, StartPayload, SuccessPayload, ErrorPayload>({
		start: state => { state.status = 'loading'; },
		success: state => { state.status = 'loaded'; },
		error: state => { state.status = 'error'; }
	});
};

export { getReducer, getApiReducers, getEntityFetchReducers, createSlice };
