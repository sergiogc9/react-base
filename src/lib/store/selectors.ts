import { useSelector as reduxUseSelector } from 'react-redux';

import { State } from 'store/types';

type Selector<T> = (state :State) => T;

const useSelector = <T>(selector: Selector<T>) => reduxUseSelector<State, T>(selector);

export { useSelector };
