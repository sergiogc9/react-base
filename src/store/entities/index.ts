import { combineReducers } from '@reduxjs/toolkit';

import selectors from './selectors';
import * as author from './Author';
import * as book from './Book';

export const reducer = combineReducers({
	author: author.reducer,
	book: book.reducer
});

export interface State {
	author: typeof author.INITIAL_STATE;
	book: typeof book.INITIAL_STATE;
}

export const INITIAL_STATE = {
	author: author.INITIAL_STATE,
	book: book.INITIAL_STATE
};

export { selectors };
