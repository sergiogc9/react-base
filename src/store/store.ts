import { combineReducers, configureStore } from '@reduxjs/toolkit';

import { customMiddlewares, sagaMiddleware } from 'middleware';
import * as auth from './auth';
import * as entities from './entities';
import * as notifications from './notifications';
import * as ui from './ui';
import sagas from './sagas';

// create reducers
const reducers = combineReducers({
	auth: auth.reducer,
	entities: entities.reducer,
	notifications: notifications.reducer,
	ui: ui.reducer
});

const store = configureStore({
	reducer: reducers,
	middleware: getDefaultMiddleware => getDefaultMiddleware().concat(...customMiddlewares)
});
const INITIAL_STATE = store.getState();

sagaMiddleware.run(sagas);

export { reducers, store, INITIAL_STATE };
