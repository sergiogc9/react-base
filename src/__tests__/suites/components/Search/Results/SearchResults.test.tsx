import React from "react";
import { mount, shallow } from "enzyme";
import { RouterState } from "connected-react-router";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import merge from "lodash/merge";
import CircularProgress from "react-md/lib/Progress/CircularProgress";

import SearchResultsComponent from "@src/components/Search/Results/SearchResults";
import { INITIAL_STATE } from "@src/store";
import SearchResults from "@src/components/Search/Results";
import DocumentPlaceHolder from "@src/components/common/Document/PlaceHolder/DocumentPlaceHolder";
import DocumentItem from "@src/components/common/Document/Item/DocumentItem";
import { ApiSearchDocument, DocumentObject, ApiArticleSearchDocument, ApiPreviewSearchDocument } from "@src/class/Document";
import { UserObject } from "@src/class/User";

// TODO: Use user from test helper
const user: UserObject = ({
	id: "123456",
	name: "user",
	email: "user@user.com",
	role: "admin",
	permissions: ["document.categorize", "document.tag", "document.delete"]
});

// TODO: Use document from test helper
const baseDocument: DocumentObject = {
	place: null,
	date_from_provider: "",
	tags: null,
	queries: [],
	category: null,
	media: null,
	image_url: "",
	image: null,
	page_number: "",
	brand_associated: "",
	category_id: "",
	country: null,
	issue_number: "",
	page_occupation: 0,
	author: {
		id: 1506856080,
		name: "Igor Giannasi"
	},
	content: "content",
	date: "2019-04-04T06:30:57Z",
	id: "15543594571385654087",
	language: {
		code: "pt",
		id: "7",
		name: "Portuguese"
	},
	link: "http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==",
	location: {
		continent: {
			id: "02",
			name: "South America"
		},
		country: {
			id: "0208",
			iso: "BR",
			name: "Brazil"
		}
	},
	provider: "News",
	rank: {
		audience: 1311327,
		similarweb_monthly_visits: 61423055
	},
	social: {},
	source: {
		id: "1233730921",
		title: "http://www.terra.com.br",
		url: "http://www.terra.com.br",
		name: ""
	},
	title: "Peça &#39;O Aniversário de Jean Lucca&#39; retrata a &#39;cultura da indiferença&#39; na sociedade brasileira"
};

const routerLocation: RouterState["location"] = {
	pathname: "/article",
	search: "",
	state: "",
	hash: ""
};

const mockStore = configureMockStore();

function getWrappedComponent(component: JSX.Element, stateSlice: object) {
	const store = mockStore(merge({}, INITIAL_STATE, stateSlice));
	return mount(<Provider store={store}>{component}</Provider>);
}

describe("Search Results", () => {

	it("Test component loading documents", () => {
		const wrapper = getWrappedComponent(<SearchResults />, {
			router: { location: { pathname: "/article" } },
			search: {
				results: {
					loadingDocuments: true
				}
			}
		});
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find(CircularProgress)).toHaveLength(1);
	});

	it("Test component loading article documents", () => {
		const documents: ApiArticleSearchDocument[] = [
			{ id: "123456789", source: { queries: [], categories_id: "", tags: [] } }
		];
		const wrapper = getWrappedComponent(<SearchResults />, {
			router: { location: { pathname: "/article" } },
			search: {
				results: {
					loadingDocuments: false,
					loadingDocumentsSources: true,
					documents
				}
			}
		});
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find(CircularProgress)).toHaveLength(0);
		expect(wrapper.find(DocumentPlaceHolder)).toHaveLength(1);
	});

	it("Test component loading preview documents", () => {
		const documents: ApiPreviewSearchDocument[] = ["123456789"];
		const wrapper = getWrappedComponent(<SearchResults />, {
			router: { location: { pathname: "/focusId/feed/create/socialmedia" } },
			search: {
				results: {
					loadingDocuments: false,
					loadingDocumentsSources: true,
					documents
				}
			}
		});
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find(CircularProgress)).toHaveLength(0);
		expect(wrapper.find(DocumentPlaceHolder)).toHaveLength(1);
	});

	it("Test component loading documents without documents", () => {
		const documents: ApiSearchDocument[] | null = null;
		const wrapper = getWrappedComponent(<SearchResults />, {
			router: { location: { pathname: "/article" } },
			search: {
				results: {
					loadingDocuments: false,
					loadingDocumentsSources: true,
					documents
				}
			}
		});
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find(CircularProgress)).toHaveLength(0);
		expect(wrapper.find(DocumentPlaceHolder)).toHaveLength(0);
	});

	it("Test component loading documents without documents sources", () => {
		const documentSources: DocumentObject[] | null = null;
		const wrapper = getWrappedComponent(<SearchResults />, {
			router: { location: { pathname: "/article" } },
			app: {
				profile: {
					user
				}
			},
			search: {
				results: {
					loadingDocuments: false,
					loadingDocumentsSources: false,
					documents: null,
					documentSources
				}
			}
		});
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find(CircularProgress)).toHaveLength(0);
		expect(wrapper.find(DocumentPlaceHolder)).toHaveLength(0);
		expect(wrapper.find(DocumentItem)).toHaveLength(0);
		expect(wrapper.find('#documentResultsNoResults')).toHaveLength(1);
	});

	it("Test component loading documents with document source", () => {
		const documentSources: DocumentObject[] | null = [baseDocument];
		const wrapper = getWrappedComponent(<SearchResults />, {
			router: { location: { pathname: "/article" } },
			app: {
				profile: {
					user
				}
			},
			search: {
				results: {
					loadingDocuments: false,
					loadingDocumentsSources: false,
					documents: null,
					documentSources
				}
			}
		});
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find(CircularProgress)).toHaveLength(0);
		expect(wrapper.find(DocumentPlaceHolder)).toHaveLength(0);
		expect(wrapper.find(DocumentItem)).toHaveLength(1);
	});

	it("Test component loading documents with document source", () => {
		const documentSources: any[] | null = [{ ...baseDocument, provider: "FAKE" }];
		const wrapper = getWrappedComponent(<SearchResults />, {
			router: { location: { pathname: "/article" } },
			search: {
				results: {
					loadingDocuments: false,
					loadingDocumentsSources: false,
					documents: null,
					documentSources
				}
			}
		});
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find(CircularProgress)).toHaveLength(0);
		expect(wrapper.find(DocumentPlaceHolder)).toHaveLength(0);
		expect(wrapper.find(DocumentItem)).toHaveLength(0);
	});

	it("Test component remove document action", () => {
		const documentSources: DocumentObject[] | null = [baseDocument];

		const onRemoveDocumentMock = jest.fn();
		const onChangeDocumentCheckMock = jest.fn();
		const component = mount(
			<SearchResultsComponent
				location={routerLocation}
				loadingDocumentsSources={false}
				documents={null}
				documentSources={documentSources}
				loadingDocuments={false}
				onRemoveDocument={onRemoveDocumentMock}
				loadingRemoveDocument={{}}
				pageDocuments={0}
				total={20}
				documentsChecked={{}}
				loadingSetTagsCategoryDocument={{}}
				onChangeDocumentCheck={onChangeDocumentCheckMock}
				user={user}
			/>);
		expect(onRemoveDocumentMock).toHaveBeenCalledTimes(0);
		component.find('.icon-cancel').at(0).simulate('click');
		expect(onRemoveDocumentMock).toHaveBeenCalledTimes(1);
	});

	it("Test component remove document action without id", () => {
		const documentSources: DocumentObject[] | null = [{ ...baseDocument, id: "" }];

		const onRemoveDocumentMock = jest.fn();
		const onChangeDocumentCheckMock = jest.fn();
		const component = mount(
			<SearchResultsComponent
				location={routerLocation}
				loadingDocumentsSources={false}
				documents={null}
				documentSources={documentSources}
				loadingDocuments={false}
				onRemoveDocument={onRemoveDocumentMock}
				loadingRemoveDocument={{}}
				pageDocuments={0}
				total={20}
				documentsChecked={{}}
				loadingSetTagsCategoryDocument={{}}
				onChangeDocumentCheck={onChangeDocumentCheckMock}
				user={user}
			/>);
		expect(onRemoveDocumentMock).toHaveBeenCalledTimes(0);
		component.find('.icon-cancel').at(0).simulate('click');
		expect(onRemoveDocumentMock).toHaveBeenCalledTimes(1);
	});

	it("Test component loading remove document", () => {
		const documentSources: DocumentObject[] | null = [{ ...baseDocument, id: "1234" }];

		const onRemoveDocumentMock = jest.fn();
		const onChangeDocumentCheckMock = jest.fn();
		const component = mount(
			<SearchResultsComponent
				location={routerLocation}
				loadingDocumentsSources={false}
				documents={null}
				documentSources={documentSources}
				loadingDocuments={false}
				onRemoveDocument={onRemoveDocumentMock}
				loadingRemoveDocument={{ 1234: true }}
				pageDocuments={0}
				total={20}
				documentsChecked={{}}
				loadingSetTagsCategoryDocument={{}}
				onChangeDocumentCheck={onChangeDocumentCheckMock}
				user={user}
			/>);
		expect(component.html()).toMatchSnapshot();
	});

	it("Test component select", () => {
		const documentSources: DocumentObject[] | null = [{ ...baseDocument, id: "1234" }];

		const onRemoveDocumentMock = jest.fn();
		const onChangeDocumentCheckMock = jest.fn();
		const component = mount(
			<SearchResultsComponent
				location={routerLocation}
				loadingDocumentsSources={false}
				documents={null}
				documentSources={documentSources}
				loadingDocuments={false}
				onRemoveDocument={onRemoveDocumentMock}
				loadingRemoveDocument={{}}
				pageDocuments={0}
				total={20}
				documentsChecked={{}}
				loadingSetTagsCategoryDocument={{}}
				onChangeDocumentCheck={onChangeDocumentCheckMock}
				user={user}
			/>);
		expect(onChangeDocumentCheckMock).toHaveBeenCalledTimes(0);
		const checkbox: any = component.find('Checkbox.document-results-item-select-checkbox').at(0);
		checkbox.props().onChange();
		expect(onChangeDocumentCheckMock).toHaveBeenCalledTimes(1);
	});

	it("Test categorization visibility", () => {
		const onRemoveDocumentMock = jest.fn();
		const onChangeDocumentCheckMock = jest.fn();
		const component = shallow(
			<SearchResultsComponent
				location={routerLocation}
				loadingDocumentsSources={false}
				documents={null}
				documentSources={[baseDocument]}
				loadingDocuments={false}
				onRemoveDocument={onRemoveDocumentMock}
				loadingRemoveDocument={{}}
				pageDocuments={0}
				total={20}
				documentsChecked={{}}
				loadingSetTagsCategoryDocument={{}}
				onChangeDocumentCheck={onChangeDocumentCheckMock}
				user={user}
			/>);
		expect(component.debug()).toMatchSnapshot();

		component.setProps({ ...component.props, documentSources: [{ ...baseDocument, category: "123456" }] });
		expect(component.debug()).toMatchSnapshot();

		component.setProps({ ...component.props, documentSources: [{ ...baseDocument, company: "Company" }] });
		expect(component.debug()).toMatchSnapshot();

		component.setProps({ ...component.props, documentSources: [{ ...baseDocument, tags: ["test"] }] });
		expect(component.debug()).toMatchSnapshot();
	});

	it("Test document checkbox visibility permissions", () => {
		const onRemoveDocumentMock = jest.fn();
		const onChangeDocumentCheckMock = jest.fn();
		const documentSources: DocumentObject[] = [baseDocument];
		const component = shallow(
			<SearchResultsComponent
				location={routerLocation}
				loadingDocumentsSources={false}
				documents={null}
				documentSources={documentSources}
				loadingDocuments={false}
				onRemoveDocument={onRemoveDocumentMock}
				loadingRemoveDocument={{}}
				pageDocuments={0}
				total={20}
				documentsChecked={{}}
				loadingSetTagsCategoryDocument={{}}
				onChangeDocumentCheck={onChangeDocumentCheckMock}
				user={user}
			/>);
		expect(component.find(`#documentSelect-${baseDocument.id}`).exists()).toBe(true);

		component.setProps({ ...component.props, user: { ...user, permissions: [] } });
		expect(component.find(`#documentSelect-${baseDocument.id}`).exists()).toBe(false);
	});

	it("Test remove button permissions", () => {
		const onRemoveDocumentMock = jest.fn();
		const onChangeDocumentCheckMock = jest.fn();
		const documentSources: DocumentObject[] = [baseDocument];
		const component = shallow(
			<SearchResultsComponent
				location={routerLocation}
				loadingDocumentsSources={false}
				documents={null}
				documentSources={documentSources}
				loadingDocuments={false}
				onRemoveDocument={onRemoveDocumentMock}
				loadingRemoveDocument={{}}
				pageDocuments={0}
				total={20}
				documentsChecked={{}}
				loadingSetTagsCategoryDocument={{}}
				onChangeDocumentCheck={onChangeDocumentCheckMock}
				user={user}
			/>);
		expect(component.find('.remove-document-button').exists()).toBe(true);

		component.setProps({ ...component.props, user: { ...user, permissions: [] } });
		expect(component.find('.remove-document-button').exists()).toBe(false);
	});
});
