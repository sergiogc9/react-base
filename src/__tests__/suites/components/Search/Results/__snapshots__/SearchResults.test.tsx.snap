// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results Test categorization visibility 1`] = `
"<div id=\\"results\\">
  <div id=\\"document-15543594571385654087\\" className=\\"document-results-item\\">
    <div className=\\"document-results-item-select\\">
      <Checkbox id=\\"documentSelect-15543594571385654087\\" className=\\"document-results-item-select-checkbox\\" name=\\"documentSelect\\" aria-label=\\"15543594571385654087\\" checked={false} onChange={[Function: onChange]} checkedIcon={{...}} uncheckedIcon={{...}} />
    </div>
    <DocumentPlaceHolder document={{...}} />
    <div className=\\"document-results-item-actions\\">
      <withInk(withTooltip(Button)) icon={true} data-id=\\"15543594571385654087\\" className=\\"icon-cancel remove-document-button\\" title=\\"result.remove\\" onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300} />
    </div>
  </div>
</div>"
`;

exports[`Search Results Test categorization visibility 2`] = `
"<div id=\\"results\\">
  <div id=\\"document-15543594571385654087\\" className=\\"document-results-item\\">
    <div className=\\"document-results-item-select\\">
      <Checkbox id=\\"documentSelect-15543594571385654087\\" className=\\"document-results-item-select-checkbox\\" name=\\"documentSelect\\" aria-label=\\"15543594571385654087\\" checked={false} onChange={[Function: onChange]} checkedIcon={{...}} uncheckedIcon={{...}} />
    </div>
    <DocumentPlaceHolder document={{...}} />
    <div className=\\"document-results-item-actions\\">
      <withInk(withTooltip(Button)) icon={true} data-id=\\"15543594571385654087\\" className=\\"icon-cancel remove-document-button\\" title=\\"result.remove\\" onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300} />
      <Connect(DocumentCategorization) document={{...}} />
    </div>
  </div>
</div>"
`;

exports[`Search Results Test categorization visibility 3`] = `
"<div id=\\"results\\">
  <div id=\\"document-15543594571385654087\\" className=\\"document-results-item\\">
    <div className=\\"document-results-item-select\\">
      <Checkbox id=\\"documentSelect-15543594571385654087\\" className=\\"document-results-item-select-checkbox\\" name=\\"documentSelect\\" aria-label=\\"15543594571385654087\\" checked={false} onChange={[Function: onChange]} checkedIcon={{...}} uncheckedIcon={{...}} />
    </div>
    <DocumentPlaceHolder document={{...}} />
    <div className=\\"document-results-item-actions\\">
      <withInk(withTooltip(Button)) icon={true} data-id=\\"15543594571385654087\\" className=\\"icon-cancel remove-document-button\\" title=\\"result.remove\\" onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300} />
      <Connect(DocumentCategorization) document={{...}} />
    </div>
  </div>
</div>"
`;

exports[`Search Results Test categorization visibility 4`] = `
"<div id=\\"results\\">
  <div id=\\"document-15543594571385654087\\" className=\\"document-results-item\\">
    <div className=\\"document-results-item-select\\">
      <Checkbox id=\\"documentSelect-15543594571385654087\\" className=\\"document-results-item-select-checkbox\\" name=\\"documentSelect\\" aria-label=\\"15543594571385654087\\" checked={false} onChange={[Function: onChange]} checkedIcon={{...}} uncheckedIcon={{...}} />
    </div>
    <DocumentPlaceHolder document={{...}} />
    <div className=\\"document-results-item-actions\\">
      <withInk(withTooltip(Button)) icon={true} data-id=\\"15543594571385654087\\" className=\\"icon-cancel remove-document-button\\" title=\\"result.remove\\" onClick={[Function: onClick]} inkTransitionOverlap={150} inkTransitionEnterTimeout={450} inkTransitionLeaveTimeout={300} />
      <Connect(DocumentCategorization) document={{...}} />
    </div>
  </div>
</div>"
`;

exports[`Search Results Test component loading article documents 1`] = `"<div id=\\"results\\"><div class=\\"document-place-holder\\"><div class=\\"document-place-holder-left-info\\"><div class=\\"document-place-holder-provider-icon animated\\"></div><div class=\\"document-place-holder-counters animated\\"></div></div><div class=\\"document-place-holder-content\\"><div class=\\"document-place-holder-content-title animated\\"></div><div class=\\"document-place-holder-content-source animated\\"></div></div></div></div>"`;

exports[`Search Results Test component loading documents 1`] = `"<div id=\\"results\\"><svg id=\\"documentResultsLoading\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate\\" width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg></div>"`;

exports[`Search Results Test component loading documents with document source 1`] = `"<div id=\\"results\\"><div id=\\"document-15543594571385654087\\" class=\\"document-results-item\\"><div class=\\"document-results-item-select\\"><div class=\\"md-selection-control-container document-results-item-select-checkbox\\"><input id=\\"documentSelect-15543594571385654087\\" type=\\"checkbox\\" class=\\"md-selection-control-input\\" name=\\"documentSelect\\" aria-hidden=\\"true\\" value=\\"\\"><label for=\\"documentSelect-15543594571385654087\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"15543594571385654087\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div></div><div class=\\"document-item\\"><div class=\\"document-item-left-info\\"><div class=\\"document-item-provider-icon-box\\"><i class=\\"md-icon icon-news document-item-provider-icon News\\"></i></div><div class=\\"document-item-counters\\"><div class=\\"document-item-counter-value\\"><span title=\\"Reach/Subscribers: 61.4M visitors\\">61.4M</span><i class=\\"documen-item-counter-icon reach material-icons\\" title=\\"Reach/Subscribers: 61.4M visitors\\">wifi</i></div><div class=\\"document-item-counter-value\\"><span title=\\"Engagement/Views: 0 shares\\">0</span><i class=\\"documen-item-counter-icon icon-like\\" title=\\"Engagement/Views: 0 shares\\"></i></div></div></div><div class=\\"document-item-author-info\\"><div class=\\"document-item-author-icon\\"><a href=\\"http://www.terra.com.br\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><img src=\\"https://faviewhaiy.s3.amazonaws.com/1/1233730921.ico\\" alt=\\"\\"></a></div><div class=\\"document-item-author-name\\"><a href=\\"http://www.terra.com.br\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><div>http://www.terra.com.br (BR)</div></a></div><div class=\\"document-item-date\\" title=\\"Thursday, April 4, 2019 6:30 AM\\">Apr 04</div><div class=\\"document-item-page\\"></div></div><div class=\\"document-item-content\\"><a href=\\"http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><div class=\\"document-item-content-title\\">Peça 'O Aniversário de Jean Lucca' retrata a 'cultura da indiferença' na sociedade brasileira</div></a><a href=\\"http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><div class=\\"document-item-content-source\\">content</div></a></div><div class=\\"document-item-image\\"><a href=\\"http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" type=\\"main\\" class=\\"document-item-image-link active-image\\"><img src=\\"\\" alt=\\"\\"></a></div></div><div class=\\"document-results-item-actions\\"><button data-id=\\"15543594571385654087\\" title=\\"result.remove\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block icon-cancel remove-document-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\"></i></button></div></div></div>"`;

exports[`Search Results Test component loading documents with document source 2`] = `"<div id=\\"results\\"></div>"`;

exports[`Search Results Test component loading documents without documents 1`] = `"<div id=\\"results\\"></div>"`;

exports[`Search Results Test component loading documents without documents sources 1`] = `"<div id=\\"results\\"><div id=\\"documentResultsNoResults\\">results.no_results</div></div>"`;

exports[`Search Results Test component loading preview documents 1`] = `"<div id=\\"results\\"><div class=\\"document-place-holder\\"><div class=\\"document-place-holder-left-info\\"><div class=\\"document-place-holder-provider-icon animated\\"></div><div class=\\"document-place-holder-counters animated\\"></div></div><div class=\\"document-place-holder-content\\"><div class=\\"document-place-holder-content-title animated\\"></div><div class=\\"document-place-holder-content-source animated\\"></div></div></div></div>"`;

exports[`Search Results Test component loading remove document 1`] = `"<div id=\\"results\\"><div class=\\"removing-document\\"><div class=\\"removing-document-box\\"><svg id=\\"removingDocument-1234\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate removing-document-loading\\" width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg></div><div id=\\"document-1234\\" class=\\"document-results-item\\"><div class=\\"document-results-item-select\\"><div class=\\"md-selection-control-container document-results-item-select-checkbox\\"><input id=\\"documentSelect-1234\\" type=\\"checkbox\\" class=\\"md-selection-control-input\\" name=\\"documentSelect\\" aria-hidden=\\"true\\" value=\\"\\"><label for=\\"documentSelect-1234\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"1234\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div></div><div class=\\"document-item\\"><div class=\\"document-item-left-info\\"><div class=\\"document-item-provider-icon-box\\"><i class=\\"md-icon icon-news document-item-provider-icon News\\"></i></div><div class=\\"document-item-counters\\"><div class=\\"document-item-counter-value\\"><span title=\\"Reach/Subscribers: 61.4M visitors\\">61.4M</span><i class=\\"documen-item-counter-icon reach material-icons\\" title=\\"Reach/Subscribers: 61.4M visitors\\">wifi</i></div><div class=\\"document-item-counter-value\\"><span title=\\"Engagement/Views: 0 shares\\">0</span><i class=\\"documen-item-counter-icon icon-like\\" title=\\"Engagement/Views: 0 shares\\"></i></div></div></div><div class=\\"document-item-author-info\\"><div class=\\"document-item-author-icon\\"><a href=\\"http://www.terra.com.br\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><img src=\\"https://faviewhaiy.s3.amazonaws.com/1/1233730921.ico\\" alt=\\"\\"></a></div><div class=\\"document-item-author-name\\"><a href=\\"http://www.terra.com.br\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><div>http://www.terra.com.br (BR)</div></a></div><div class=\\"document-item-date\\" title=\\"Thursday, April 4, 2019 6:30 AM\\">Apr 04</div><div class=\\"document-item-page\\"></div></div><div class=\\"document-item-content\\"><a href=\\"http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><div class=\\"document-item-content-title\\">Peça 'O Aniversário de Jean Lucca' retrata a 'cultura da indiferença' na sociedade brasileira</div></a><a href=\\"http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><div class=\\"document-item-content-source\\">content</div></a></div><div class=\\"document-item-image\\"><a href=\\"http://discover.staging.launchmetrics.com/r/d/17Xw6MAhy0c=_ckkDAAAAAAAAAAAAAAAAAA==\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" type=\\"main\\" class=\\"document-item-image-link active-image\\"><img src=\\"\\" alt=\\"\\"></a></div></div><div class=\\"document-results-item-actions\\"><button data-id=\\"1234\\" title=\\"result.remove\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block icon-cancel remove-document-button\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\"></i></button></div></div></div></div>"`;
